typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -324,20 +325,25 @@
   (define-syntax-class name-exists-kw
     (pattern #:name-exists))
   (syntax-parse stx
+    [_ #:when (eq? 'module-begin (syntax-local-context))
+       ;; it would be inconvenient to find the correct #%module-begin here, so we rely on splicing
+       #`(begin #,stx (begin))]
```

Comment:
No, this solves a different problem (which is what happens if `require/opaque-type` is the only thing in the module). The import problem is solved by the contract generation infrastructure already doing the right thing and inserting the same `require` manually into the `contract-defs` submodule.